pkg/controller/node/nodecontroller.go

```diff
@@ -575,15 +601,28 @@ func (nc *NodeController) monitorNodeStatus() error {
 		}
 	}
 	nc.handleDisruption(zoneToNodeConditions, nodes)
+	nc.updateEvictionMetric(Evictions10Minutes, nc.evictions10Minutes)
+	nc.updateEvictionMetric(Evictions1Hour, nc.evictions1Hour)
 
 	return nil
 }
 
+func (nc *NodeController) updateEvictionMetric(metric *prometheus.GaugeVec, data *evictionData) {
+	data.slideWindow()
+	zones := data.getZones()
+	for _, z := range zones {
+		metric.WithLabelValues(z).Set(float64(data.countEvictions(z)))
+	}
+}
+
 func (nc *NodeController) handleDisruption(zoneToNodeConditions map[string][]*api.NodeCondition, nodes *api.NodeList) {
 	newZoneStates := map[string]zoneState{}
 	allAreFullyDisrupted := true
 	for k, v := range zoneToNodeConditions {
-		newState := nc.computeZoneStateFunc(v)
+		ZoneSize.WithLabelValues(k).Set(float64(len(v)))
+		unhealthy, newState := nc.computeZoneStateFunc(v)
+		ZoneHealth.WithLabelValues(k).Set(float64(100*(len(v)-unhealthy)) / float64(len(v)))
```

Comment:
Yes, if it were 0, then there would be no Nodes in the given zone, and hence the zone would not exist. This map is created from scratch each time `monitorNodeStatus` is called.